File: analysis/source/descriptive/descriptive.m
clear all

PATHS = ReadYaml('config_global.yaml');
data = dataset('File', sprintf('%s/data_matlab.txt', PATHS.build.prepare_data));

fid = fopen(sprintf('%s/table_matlab.txt', PATHS.build.descriptive), 'wt');
fprintf(fid, '<tab:table>\n');
fprintf(fid, '%.1f\n%.3f\n%g\n%g', mean(data.x), std(data.x), max(data.x), min(data.x));
fclose(fid);

set(figure, 'visible', 'off');
hist(data.x);
print(sprintf('%s/plot_matlab.eps', PATHS.build.descriptive), '-depsc');

exit

File: analysis/source/descriptive/descriptive.r
library(yaml)

main <- function() {
    PATHS <- yaml.load_file("config_global.yaml")
    data  <- read.table(sprintf("%s/data.txt", PATHS$build$prepare_data), header = TRUE)
    
    sink(sprintf("%s/table_r.txt", PATHS$build$descriptive))
    cat("<tab:table>\n")
    cat(sprintf("%s\n%.3f\n%d\n%d",
               mean(data$x), sd(data$x), max(data$x), min(data$x)))
    sink()

    postscript(sprintf("%s/plot_r.eps", PATHS$build$descriptive))
    hist(data$x, breaks = 10)
    dev.off()
}

main()
